pkg/kubectl/cmd/expose.go

```diff
@@ -261,7 +261,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 		info.Refresh(object, true)
 		if cmdutil.GetDryRunFlag(cmd) {
-			return f.PrintObject(cmd, mapper, object, out)
+			if len(cmdutil.GetFlagString(cmd, "output")) > 0 {
+				return f.PrintObject(cmd, mapper, object, out)
+			}
+			cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, true, "exposed")
```

Comment:
Hm, old behavior prints the resource that was exposed no matter what the specified `--output` was; this one keeps that behvavior when an `--output` other than the standard one is specified, but outputs  `<resource_kind> "<resource>" exposed (DRY RUN)` otherwise, in order to provide a clearer output. This will match the rest of the `dry-run` outputs